### PR TITLE
[incubator-kie-issues-1131] test migration from V7 to code generation-7

### PR DESCRIPTION
**Pull request by Abhitocode**
In addition to previous changes associated,
Migrated test case `testErrorBetweenProcessesProcess` but the test seems to have issue where expected state of sub process should be aborted but we are getting state pending

The test can be identified by referring to Activitytest.java:
https://github.com/apache/incubator-kie-kogito-runtimes/blob/main/jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/ActivityTest.java

Closes https://github.com/apache/incubator-kie-issues/issues/1131.

NOTE : This pr contains test which is yet to be resolved, commented the issues for particular tests, will update and push once resolved.

### CODE REVIEW

**Comment by kie-ci3** [comment]
**PR job** `#1` was: **UNSTABLE**
Possible explanation: This should be test failures


<details>
<summary><b>Reproducer</b></summary>



build-chain build full_downstream  -f 'https://raw.githubusercontent.com/${AUTHOR:apache}/incubator-kie-kogito-pipelines/${BRANCH:main}/.ci/buildchain-config-pr-cdb.yaml' -o 'bc' -p apache/incubator-kie-kogito-runtimes -u https://github.com/apache/incubator-kie-kogito-runtimes/pull/3583 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3583/1/display/redirect

**Test results:**
- PASSED: 3263
- FAILED: 2

Those are the test failures: 
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3583/1/testReport/org.jbpm.bpmn2/ActivityTest/testErrorBetweenProcessesProcess/">org.jbpm.bpmn2.ActivityTest.testErrorBetweenProcessesProcess</a></summary>
<br/>expected: 3<br/> but was: 0
</details>
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-runtimes-pr/job/PR-3583/1/testReport/io.quarkus.it.kogito.devmode/DevMojoIT/testDRLHotReload/">io.quarkus.it.kogito.devmode.DevMojoIT.testDRLHotReload</a></summary>
Condition with io.quarkus.it.kogito.devmode.DevMojoIT was not fulfilled within 2 minutes.
</details>

**Comment by elguardian** [comment]
Hi the test is not correct. You need to access the process instance created by the engine, not your process instance like you are doing in here.
For doing so, you need to create a listener and check the event start process (at that moment you can fetch the new process instance created and matched against the the subprocess id.

File: jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/ActivityTest.java [hunk 1]
```diff
@@ -1246,21 +1250,23 @@ public void testServiceTaskWithMvelJaxbTransformation() throws Exception {
...
+        org.kie.kogito.process.Process<ErrorsBetweenProcessModel> process = ErrorsBetweenProcessProcess.newProcess(app);
+        ErrorsBetweenProcessModel model = process.createModel();
+        model.setTipoEvento("error");
+        model.setPasoVariable(3);
+        ProcessInstance<ErrorsBetweenProcessModel> processInstance = process.createInstance(model);
+        processInstance.start();
+        assertThat(processInstance.status()).isEqualTo(ProcessInstance.STATE_COMPLETED);
+        //expected state of sub process should be aborted. But we are getting state pending
+        assertThat(subProcessInstance.status()).isEqualTo(ProcessInstance.STATE_ABORTED);
+        Assertions.assertEquals("error desde Subproceso", processInstance.variables().getEvent());
```
Comment (pibizza) [inline jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/ActivityTest.java:1269]:
```suggestion
       assertThat(processInstance.variables().getEvent()).isEqualTo("error desde Subproceso");
```

**Comment by Abhitocode** [comment]
@elguardian I have implemented the listener and registered, that allows me to get the sub-process instance of type '**org.kie.api.runtime.process.ProcessInstance**', however in order to check the status in migrated test I need sub-process instance of type '**org.kie.kogito.process.ProcessInstance<org.jbpm.bpmn2.subprocess.ErrorsBetweenSubProcessModel>**' . An alternative solution might be to get the process instance by the id (which we can obtain from listener event) if there is any way to do so. But however I'm not sure how to retrieve the process instance by id.

This is how I implemented and registered the listener :

`DefaultKogitoProcessEventListener listener = new DefaultKogitoProcessEventListener() {
            @Override
            public void afterProcessStarted(ProcessStartedEvent event) {
                subProcessInstanceList.add((ProcessInstance<ErrorsBetweenSubProcessModel>) event.getProcessInstance());
            }
        };
        ProcessTestHelper.registerProcessEventListener(app, listener);
`

**Comment by elguardian** [comment]
event.getProcessInstance().getId()

**Comment by Abhitocode** [comment]
Hi @elguardian Can you guide me on how to obtain the org.kie.kogito.process.ProcessInstance<org.jbpm.bpmn2.subprocess.ErrorsBetweenSubProcessModel> object from the event object using a listener? Additionally, is the process instance id the same in both kie api and kogito api? If so, how can we retrieve the org.kie.kogito.process.ProcessInstance<org.jbpm.bpmn2.subprocess.ErrorsBetweenSubProcessModel> object using that id or is there any other alternative to get the sub-process instance?

**Comment by elguardian** [comment]
@Abhitocode yes the id is the same. you can access the ProcessInstances from the definition to get the process instance

**Comment by Abhitocode** [comment]
Hi, I've implemented a ProcessEventListener to capture the process instance created by the engine, but I'm facing an issue. The beforeProcessStarted and afterProcessStarted methods of the listener are not being executed. Can you please help me understand if I'm missing something in the configuration or implementation?

`  Application app = ProcessTestHelper.newApplication();
        org.kie.kogito.process.Process<ErrorsBetweenSubProcessModel> errorsBetweenSubProcessProcess = ErrorsBetweenSubProcessProcess.newProcess(app);
        /*
         * ProcessInstance<ErrorsBetweenSubProcessModel> subProcessInstance = errorsBetweenSubProcessProcess
         * .createInstance(errorsBetweenSubProcessProcess.createModel());
         */
        org.kie.kogito.process.Process<ErrorsBetweenProcessModel> process = ErrorsBetweenProcessProcess.newProcess(app);
        final Set<String> processInstanceIdSet = new HashSet<>();
        DefaultKogitoProcessEventListener listener = new DefaultKogitoProcessEventListener() {
            @Override
            public void beforeProcessStarted(ProcessStartedEvent event) {
                logger.info("sub process id: {}, process id: {}", errorsBetweenSubProcessProcess.id(), process.id());
                logger.info("Process Instance process id: {}, instance id: {}", event.getProcessInstance().getProcessId(), event.getProcessInstance().getId());

                // Check if the process instance is the one we're interested in
                if (event.getProcessInstance().getProcessId().equals(errorsBetweenSubProcessProcess.id())) {
                    processInstanceIdSet.add(event.getProcessInstance().getId());
                }
            }
        };
        ProcessTestHelper.registerProcessEventListener(app, listener);`

**Comment by elguardian** [comment]
ActivityTest.testErrorBetweenProcessesProcess failures related. please fix

**Comment by Abhitocode** [comment]
Il be submitting this particular test **testErrorBetweenProcessesProcess** in the upcoming pr related to DataTest.java. Removing it from this as of now.

**Pull request closed without merging**